tests/integration/models/multifidelity/test_multifidelity_models.py

```diff
@@ -0,0 +1,574 @@
+
+    search_space = Box([0.0], [10.0])
+
+    model = MultifidelityAutoregressive(
+        build_multifidelity_autoregressive_models(
+            dataset, num_fidelities=2, input_search_space=search_space
+        )
+    )
+
+    model.update(dataset)
+    model.optimize(dataset)
+
+    kernel = gpflow.kernels.Matern52(variance=1e-18, lengthscales=[0.2, 0.2])
+    gpr = gpflow.models.GPR(dataset.astuple(), kernel, noise_variance=1e-5)
+
+    model.lowest_fidelity_signal_model = GaussianProcessRegression(gpr)
+
+    # Add some low fidelity points to check that predict on different fids works
+    test_locations_30 = tf.Variable(np.linspace(0, 10, 30), dtype=tf.float64)[:, None]
+    lf_test_locations = add_fidelity_column(test_locations_30, 0)
+    test_locations_32 = tf.Variable(np.linspace(0, 10, 32), dtype=tf.float64)[:, None]
+    hf_test_locations = add_fidelity_column(test_locations_32, 1)
+
+    concatenated_test_locations = tf.concat([lf_test_locations, hf_test_locations], axis=0)
+
+    concat_prediction, _ = model.predict(concatenated_test_locations)
+    hf_prediction = concat_prediction[30:]
+    hf_prediction_direct, _ = model.fidelity_residual_models[1].predict(test_locations_32)
+
+    npt.assert_allclose(hf_prediction, hf_prediction_direct)
+
+
+def test_multifidelity_autoregressive_predict_hf_is_consistent_when_hf_residual_is_flat() -> None:
+
+    xs_low = tf.Variable(np.linspace(0, 10, 100), dtype=tf.float64)[:, None]
+    xs_high = tf.Variable(np.linspace(0, 10, 10), dtype=tf.float64)[:, None]
+    lf_obs = tf.sin(xs_low)
+    hf_obs = 2 * tf.sin(xs_high) + tf.random.normal(
+        xs_high.shape, mean=0, stddev=1e-1, dtype=tf.float64
+    )
+
+    lf_query_points = add_fidelity_column(xs_low, 0)
+    hf_query_points = add_fidelity_column(xs_high, 1)
+
+    lf_dataset = Dataset(lf_query_points, lf_obs)
+    hf_dataset = Dataset(hf_query_points, hf_obs)
+
+    dataset = lf_dataset + hf_dataset
+
+    search_space = Box([0.0], [10.0])
+
+    model = MultifidelityAutoregressive(
+        build_multifidelity_autoregressive_models(
+            dataset, num_fidelities=2, input_search_space=search_space
+        )
+    )
+
+    model.update(dataset)
+    model.optimize(dataset)
+
+    kernel = gpflow.kernels.Matern52(variance=1e-18, lengthscales=[0.2, 0.2])
+    gpr = gpflow.models.GPR(dataset.astuple(), kernel, noise_variance=1e-5)
+
+    model.fidelity_residual_models[1] = GaussianProcessRegression(gpr)  # type: ignore
+
+    test_locations = tf.Variable(np.linspace(0, 10, 32), dtype=tf.float64)[:, None]
+    hf_test_locations = add_fidelity_column(test_locations, 1)
+
+    hf_prediction, _ = model.predict(hf_test_locations)
+    hf_prediction_direct = (
+        model.rho[1] * model.lowest_fidelity_signal_model.predict(test_locations)[0]
+    )
+
+    npt.assert_allclose(hf_prediction, hf_prediction_direct)
+
+
+def test_multifidelity_autoregressive_sample_low_fidelity_are_consistent() -> None:
```

Comment:
do we need this test if we have test_multifidelity_autoregressive_sample_lf_are_consistent_with_multiple_fidelities?